odoo/addons/base/tests/test_views.py

```diff
@@ -866,6 +866,369 @@ def test_modifiers(self):
         # implemeted elsewhere...
         modifiers_tests()
 
+    @mute_logger('odoo.addons.base.ir.ir_ui_view')
+    def test_invalid_field(self):
+        with self.assertRaises(ValidationError):
+            self.View.create({
+                'name': 'invalid field',
+                'model': 'ir.ui.view',
+                'arch': """
+                    <form string="View">
+                        <field name="name"/>
+                        <field name="not_a_field"/>
+                    </form>
+                """,
+            })
+
+    @mute_logger('odoo.addons.base.ir.ir_ui_view')
+    def test_invalid_subfield(self):
+        with self.assertRaises(ValidationError):
+            self.View.create({
+                'name': 'invalid subfield',
+                'model': 'ir.ui.view',
+                'arch': """
+                    <form string="View">
+                        <field name="name"/>
+                        <field name="inherit_children_ids">
+                            <tree name="Children">
+                                <field name="name"/>
+                                <field name="not_a_field"/>
+                            </tree>
+                        </field>
+                    </form>
+                """,
+            })
+
+    @mute_logger('odoo.addons.base.ir.ir_ui_view')
+    def test_context_in_view(self):
+        self.View.create({
+            'name': 'valid context',
+            'model': 'ir.ui.view',
+            'arch': """
+                <form string="View">
+                    <field name="name"/>
+                    <field name="model"/>
+                    <field name="inherit_id" context="{'stuff': model}"/>
+                </form>
+            """,
+        })
+        with self.assertRaises(ValidationError):
+            self.View.create({
+                'name': 'invalid context',
+                'model': 'ir.ui.view',
+                'arch': """
+                    <form string="View">
+                        <field name="name"/>
+                        <field name="inherit_id" context="{'stuff': model}"/>
+                    </form>
+                """,
+            })
+
+    @mute_logger('odoo.addons.base.ir.ir_ui_view')
+    def test_context_in_subview(self):
```

Comment:
I would have added a test if model is not in the subview but in the main view.

Reply:
:+1: